### PR TITLE
fix auto-scaling for gpu and default node group

### PR DESCRIPTION
**Pull request by tmisch**
#164 broke auto-scaling for _gpu_ and _default_ node group. It has turned out, that it is sufficient to place the tag `resources/ephemeral-storage` on the auto scaling groups. Setting the tag also in the `autoDiscovery.tags` of the cluster-autoscaler lets the cluster-autoscaler manage only those auto scaling groups that have **all** of these tags. But the cluster-autoscaler works without this new tag in `autoDiscovery.tags`.

Nevertheless, it was decided to have the  `resources/ephemeral-storage` on all auto scaling groups, since it might happen every day, that a change, e.g. a new kind of pod, requires it.

### CODE REVIEW

File: k8s.tf [hunk 1]
```diff
@@ -61,6 +66,17 @@ data "aws_eks_node_group" "gpuivsnodes" {
   node_group_name = replace(module.eks.managed_node_groups[0]["gpuivsnodes"]["managed_nodegroup_id"][0], "${local.infrastructurename}:", "")
 }
 
+resource "aws_autoscaling_group_tag" "default_node-template_resources_ephemeral-storage" {
+  autoscaling_group_name = data.aws_eks_node_group.default.resources[0].autoscaling_groups[0].name
+
+  tag {
+    key   = "k8s.io/cluster-autoscaler/node-template/resources/ephemeral-storage"
+    value = "${var.linuxNodeDiskSize}G"
```
Comment (Johannesm299) [inline k8s.tf]: nicht "Gi" ? In der Variablenbeschreibung steht zumindest "... in GiB"

File: k8s.tf [hunk 2]
```diff
@@ -97,6 +113,18 @@ resource "aws_autoscaling_group_tag" "gpuexecnodes" {
   }
 }
 
+resource "aws_autoscaling_group_tag" "gpuexecnodes_node-template_resources_ephemeral-storage" {
+  count                  = var.gpuNodePool ? 1 : 0
+  autoscaling_group_name = data.aws_eks_node_group.gpuexecnodes[0].resources[0].autoscaling_groups[0].name
+
+  tag {
+    key   = "k8s.io/cluster-autoscaler/node-template/resources/ephemeral-storage"
+    value = "${var.gpuNodeDiskSize}G"
```
Comment (Johannesm299) [inline k8s.tf]: Hier genauso wahrscheinlich "Gi" statt "G"

**Review by Johannesm299 — CHANGES_REQUESTED** [review]
Apart from ephemeral storage unit, looks good.

File: k8s.tf (hunk 1 above)
Comment (tmisch) [reply k8s.tf]: Laut https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#auto-discovery-setup ist "G" richtig.

File: k8s.tf (hunk 2 above)
Comment (tmisch) [reply k8s.tf]: Laut https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/README.md#auto-discovery-setup ist "G" korrekt ... und es funktioniert ja auch.

**Pull request merged**